### PR TITLE
🐛 ROSA: Fix version gate Ack for rosa hcp y-stream version upgrade

### PR DESCRIPTION
**Pull request by serngawy**
**What type of PR is this?**
Bug

**What this PR does / why we need it**:
This PR approve the gate version acknowledgment for ROSA-HCP y-stream upgrade 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] includes [emojis](https://github.com/kubernetes-sigs/kubebuilder-release-tools?tab=readme-ov-file#kubebuilder-project-versioning)
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Add versionGate Acknowledgement for upgrading ROSA-HCP y-stream versions
```

### CODE REVIEW

**Comment by serngawy** [comment]
@muraee @damdo please approve for test

**Review by damdo — COMMENTED** [review]
/lgtm
/assign @richardcase @muraee 


/hold

Until e2e jobs on CI are fixed.

**Comment by damdo** [comment]
@serngawy could you please add a Release Note in the description of the PR for this one? Thanks!

File: pkg/rosa/versions.go
Comment (jnpacker) [inline pkg/rosa/versions.go:56]: Typo. "acknowledeg "

**Comment by jnpacker** [comment]
Can you confirm that we are introducing the user flow that we will always require an Acknowledgement, (unless AlwaysAcknowledge) is set, for all Y-Stream upgrades (that's what the code flow looks like to me. This is being done even though OCP may NOT require it. OCP 4.16 to 4.17 does not require an acknowledge. 

@serngawy can you confirm this is also ROSA-HCP's use case, or are they aware of when to require the acknowledgement?

**Review by jnpacker — CHANGES_REQUESTED** [review]
Fix message, and question on whether this flow is now forced for all Y stream upgrades.

**Comment by serngawy** [comment]
> Can you confirm that we are introducing the user flow that we will always require an Acknowledgement, (unless AlwaysAcknowledge) is set, for all Y-Stream upgrades (that's what the code flow looks like to me. This is being done even though OCP may NOT require it. OCP 4.16 to 4.17 does not require an acknowledge.
> 
> @serngawy can you confirm this is also ROSA-HCP's use case, or are they aware of when to require the acknowledgement?

The code check first if there is an versionGate Acknowledge is required, if upgrading y-stream doesn't require Acknowledge the upgrade will proceed similar to z-stream upgrade. 

The workflow is; 
1- User set the version to upgrade  (assuming the user not aware that acknowledgement is required)
2- The ROSA-HCP will return back with error message that an acknowledgement is required to proceed with upgrade.
3- The user set the versionGate to acknowledge 
4- The upgrade will proceed 
5- once the upgrade done the version gate will set back to waitForAcknowledge.

**Review by jnpacker — APPROVED** [review]
Got an explanation and a look at the code as to why ACK doesn't apply when not required.

File: pkg/rosa/versions.go
Comment (serngawy) [reply pkg/rosa/versions.go]: done

File: pkg/rosa/versions.go
Comment (andreadecorte) [inline pkg/rosa/versions.go:56]: you should expose all the version gates that needs acknowledge here, not only the first one. Otherwise the user may be thinking that there is only one gate he will acknoledge

Comment (serngawy) [reply pkg/rosa/versions.go]: done. format the output and add all version gates

**Comment by andreadecorte** [comment]
lgtm

**Review by damdo — COMMENTED** [review]
/lgtm

**Comment by damdo** [comment]
@nrb @richardcase any of you able to do an approver type review? TY

**Comment by nrb** [comment]
CI's fixed.

/unhold

**Comment by nrb** [comment]
/test pull-cluster-api-provider-aws-test

Failed due to CI account limits, not this code.

**Pull request merged**
